filters/include/pcl/filters/spherical_voxel_grid.h

```diff
@@ -0,0 +1,274 @@
+        * \param[in] limit_min the minimum allowed field value
+        * \param[in] limit_max the maximum allowed field value
+        */
+      inline void
+      setFilterLimits (const float &limit_min, const float &limit_max)
+      {
+        filter_limit_min_ = limit_min;
+        filter_limit_max_ = limit_max;
+      }
+
+      /** \brief Get the field filter limits (min/max) set by the user. The default values are -FLT_MAX, FLT_MAX.
+        * \param[out] limit_min the minimum allowed field value
+        * \param[out] limit_max the maximum allowed field value
+        */
+      inline void
+      getFilterLimits (float &limit_min, float &limit_max) const
+      {
+        limit_min = filter_limit_min_;
+        limit_max = filter_limit_max_;
+      }
+
+      /** \brief Set to true if we want to return the data outside the interval specified by setFilterLimits (min, max).
+        * Default: false.
+        * \param[in] limit_negative return data inside the interval (false) or outside (true)
+        */
+      inline void
+      setFilterLimitsNegative (const bool limit_negative) { filter_limit_negative_ = limit_negative; }
+
+      /** \brief Get whether the data outside the interval (min/max) is to be returned (true) or inside (false).
+        * \return true if data \b outside the interval [min; max] is to be returned, false otherwise
+        */
+      inline bool
+      getFilterLimitsNegative () const { return (filter_limit_negative_); }
+
+  protected:
+      /** \brief The radial size of the voxels. */
+      float leaf_size_r_;
+
+      /** \brief The number of radial divisions. */
+      unsigned int leaf_r_divisions_;
+
+      /** \brief The number of vertical divisions. */
+      unsigned int leaf_theta_divisions_;
+
+      /** \brief The number of horizontal divisions. */
+      unsigned int leaf_phi_divisions_;
+
+      /** \brief The size of the vertical divisions (radians). */
+      float leaf_size_theta_;
+
+      /** \brief The size of the horizontal divisions (radians). */
+      float leaf_size_phi_;
+
+      /** \brief Set to true if all fields need to be downsampled, or false if just XYZ. */
+      bool downsample_all_data_;
+
+      /** \brief Minimum number of points per voxel for the centroid to be computed */
+      unsigned int min_points_per_voxel_;
+
+      /** \brief The desired user filter field name. */
+      std::string filter_field_name_;
+
+      /** \brief The minimum allowed filter value a point will be considered from. */
+      float filter_limit_min_;
+
+      /** \brief The maximum allowed filter value a point will be considered from. */
+      float filter_limit_max_;
+
+      /** \brief Set to true if we want to return the data outside (\a filter_limit_min_;\a filter_limit_max_). Default: false. */
+      bool filter_limit_negative_;
+
+      /** Cartesian origin of the spherical coordinate system. */
```

Comment:
I don't think it's necessary because we have [`JAVADOC_AUTOBRIEF=ON`](https://github.com/PointCloudLibrary/pcl/blob/master/doc/doxygen/doxyfile.in#L31).